config.go

```diff
@@ -307,14 +298,24 @@ func NewSettings(filename string) (*Settings, error) {
 // GetConfig returns the Evergreen config document. If no document is
 // present in the DB, it will return the defaults.
 // Use Settings() to get the cached settings object.
-func GetConfig() (*Settings, error) { return BootstrapConfig(GetEnvironment()) }
+func GetConfig() (*Settings, error) {
+	ctx, cancel := GetEnvironment().Context()
```

Comment:
This was kind of a band-aid defer migrating `GetConfig` to a later PR. For now we could `GetConfigContext` to pass the context, but otherwise it'll still be a child of the global context.